fmriprep/interfaces/images.py

```diff
@@ -370,7 +373,14 @@ def _run_interface(self, runtime):
 
         # Both match, qform valid (implicit with match), codes okay -> do nothing, empty report
         if matching_affines and qform_code > 0 and sform_code > 0:
-            self._results['out_file'] = self.inputs.in_file
+            if self.inputs.remove_rotation_and_shear:
+                out_fname = fname_presuffix(self.inputs.in_file, suffix='_valid',
+                                            newpath=runtime.cwd)
+                img = remove_rotation_and_shear(img)
```

Comment:
Could it be that we're not correcting to closest canonical before removing rotation and shear in this interface?

Reply:
https://circleci.com/workflow-run/a814ff84-9ddd-4d8b-9961-81348b8fa202

Reply:
So I think if we're removing rotation and shear, we need to be updating to RAS. Perhaps it makes sense to actually put that step directly into `remove_rotation_and_shear`, to prevent the case where we destroy orientation information.

If the output masks are valid, just in a different space, then we should update the targets.